packages/pytest-simcore/src/pytest_simcore/docker_swarm.py

```diff
@@ -255,38 +250,26 @@ def docker_stack(
     # - notice that the timeout is set for all services in both stacks
     # - TODO: the time to deploy will depend on the number of services selected
     try:
-        if sys.version_info >= (3, 7):
-            from tenacity._asyncio import AsyncRetrying
-
-            async def _check_all_services_are_running():
-                async for attempt in AsyncRetrying(
-                    wait=wait_fixed(1),
-                    stop=stop_after_delay(8 * MINUTE),
-                    before_sleep=before_sleep_log(log, logging.INFO),
-                    reraise=True,
-                ):
-                    with attempt:
-                        await asyncio.gather(
-                            *[
-                                asyncio.get_event_loop().run_in_executor(
-                                    None, assert_service_is_running, service
-                                )
-                                for service in docker_client.services.list()
-                            ]
-                        )
-
-            asyncio.run(_check_all_services_are_running())
+        from tenacity._asyncio import AsyncRetrying
 
-        else:
-            for attempt in Retrying(
-                wait=wait_fixed(5),
+        async def _check_all_services_are_running():
+            async for attempt in AsyncRetrying(
+                wait=wait_fixed(1),
                 stop=stop_after_delay(8 * MINUTE),
                 before_sleep=before_sleep_log(log, logging.INFO),
                 reraise=True,
             ):
                 with attempt:
-                    for service in docker_client.services.list():
-                        assert_service_is_running(service)
+                    await asyncio.gather(
+                        *[
+                            asyncio.get_event_loop().run_in_executor(
+                                None, assert_service_is_running, service

```

Comment:
Now that I look at this `assert_service_is_running` is blocking or is it just me? It's not even a coroutine.

Reply:
It is to parallelize check. It does not need to be a coroutine to run in the executor ...  SEE https://docs.python.org/3/library/asyncio-eventloop.html#executing-code-in-thread-or-process-pools